rmw_connext_shared_cpp/src/node.cpp

```diff
@@ -72,16 +73,18 @@ create_node(
   participant_qos.participant_name.name = DDS::String_dup(name);
   // since the participant name is not part of the DDS spec
   // the node name is also set in the user_data
-  size_t length = strlen(name) + strlen("name=;") +
-    strlen(namespace_) + strlen("namespace=;") +
-    strlen(context->options.security_context) + strlen("securitycontext=;") + 1;
+  size_t length = std::snprintf(
+    nullptr,
+    0,
+    "name=%s;namespace=%s;securitycontext=%s;",
+    name, namespace_, context->options.security_context) + 1;
   bool success = participant_qos.user_data.value.length(static_cast<DDS::Long>(length));
   if (!success) {
     RMW_SET_ERROR_MSG("failed to resize participant user_data");
     return NULL;
   }
 
-  int written = snprintf(
+  int written = std::snprintf(
```

Comment:
@ivanpauno uh, why the added prefix?

Reply:
Also, I just remembered that `rcutils` has a portable [`snprintf`](https://github.com/ros2/rcutils/blob/af3e14543ffa7e8d4d9d38f9c242548f76accd89/include/rcutils/snprintf.h#L29-L58) flavor.

Reply:
> @ivanpauno uh, why the added prefix?

it's not really needed. I can delete it again.

> Also, I just remembered that rcutils has a portable snprintf flavor.

This seems to be portable, as it's currently working fine.
Do you know the rationale of having a `rcutils` flavor?

Reply:
Not surprisingly, there's something peculiar about using `snprintf` in Windows. But @wjwwood may have deeper insight.

Reply:
I mean, this code is passing CI on Windows with `master`, which does the same thing.
If CI is green, I wouldn't change anything else.

Reply:
Fair enough. I won't block on it, though now I'm curious about `rcutils_snprintf`.

Reply:
Maybe the std:: avoids it but on Windows it will usually generate a compiler warning if you try to use snprintf directly. Or at least that’s my recollection.